### PR TITLE
Logging fix

### PR DESCRIPTION
**Pull request by sthapa**
Fixes for the logging from the servicex_app. It's currently running on testing-4

### CODE REVIEW

**Comment by sthapa** [comment]
Should be ready for review now @ivukotic

**Review by ivukotic — APPROVED** [review]
LGTM

File: migrations/env.py [hunk 1]
```diff
@@ -1,5 +1,3 @@
-from __future__ import with_statement
```
Comment (BenGalewsky) [inline migrations/env.py:1]: What does this have to do with logging?

File: requirements.txt [hunk 2]
```diff
@@ -6,6 +6,7 @@ pika
 flask-restful
 flask-jwt-extended>=3.0,<4.0
 passlib
+MarkupSafe == 2.0.1
```
Comment (BenGalewsky) [inline requirements.txt]: This is now in develop branch I hope

File: servicex/__init__.py [hunk 3]
```diff
@@ -43,6 +45,29 @@
...
+    def format(self, record: logging.LogRecord) -> str:
+        """
+        Format record with request id if present, otherwise assume None
+
+        :param record: LogRecord
+        :return: formatted log message
+        """
+
+        if hasattr(record, "requestId"):
+            return super().format(record)
```
Comment (BenGalewsky) [inline servicex/__init__.py]: Move this duplicated line outside of `if` statement

File: servicex/__init__.py [hunk 4]
```diff
@@ -71,12 +96,42 @@ def create_app(test_config=None,
...
+    formatter = ServiceXFormatter('%(levelname)s ' +
+                                  f"{instance} servicex_app " +
+                                  '%(requestId)s %(message)s')
+    servicex_handler = logging.StreamHandler()
+    servicex_handler.setFormatter(formatter)
+    if app.debug:
+        servicex_handler.setLevel('DEBUG')
+        app.logger.level = logging.DEBUG
+    else:
+        levels = {
```
Comment (BenGalewsky) [inline servicex/__init__.py:109]: Looks like `setLevel` [accepts strings](https://docs.python.org/3/library/logging.html#logging.Logger.setLevel) so this dict is not needed

**Review by BenGalewsky — CHANGES_REQUESTED** [review]
This is a great improvement!
There are a couple of things that don't belong in this PR and a couple of style changes, but otherwise a terrific contribution

File: requirements.txt (hunk 2 above)
Comment (sthapa) [reply requirements.txt]: Probably, I'll drop this

File: migrations/env.py (hunk 1 above)
Comment (sthapa) [reply migrations/env.py:1]: It's not related to logging per se, but it's something that hasn't been needed since python 2.6.

File: servicex/__init__.py (hunk 3 above)
Comment (sthapa) [reply servicex/__init__.py]: Updated

File: servicex/__init__.py (hunk 4 above)
Comment (sthapa) [reply servicex/__init__.py:109]: I made a slight change to fix a related bug but I still need to set app.logger.level based on the env so it's either using the dict or a bunch of if/else statements.

**Comment by sthapa** [comment]
I looked at the codecov changes and that is due to logging statements not getting coverage which I think is ok

File: migrations/versions/79d404108ebe_multipath_support.py [hunk 5]
```diff
@@ -0,0 +1,38 @@
+"""multipath support
```
Comment (BenGalewsky) [inline migrations/versions/79d404108ebe_multipath_support.py]: Please clean up this PR - this has already been merged into dev as part of the multipath file fix

**Review by BenGalewsky — CHANGES_REQUESTED** [review]
Please rebase this off of dev. Let me know if you need git help doing this

**Review by BenGalewsky — APPROVED** [review]
Thanks for fixing up the commits
